web/controllers/character_controller.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Yggdrasil.CharacterController do
+  use Yggdrasil.Web, :controller
+
+  alias Yggdrasil.Character
+
+  def index(conn, %{"filter" => %{"user_id" => user_id}}) do
+    # this should not happen, but could easily
+    # probably don't need to crash but rather return an error
+    ^user_id = conn.assigns.user
+
+    chars = Repo.all from c in Character,
+                    where: c.user_id == ^user_id,
+                    select: c
+
+    render conn, :show, data: chars
+  end
+
+  def show(conn, %{"char_id" => char_id}) do
+    user_id = conn.assigns.user
+
+    # returns nil if not found
+    # need to sort out what we want here.
+    char = Repo.one from c in Character,
```

Comment:
Same kind of deal, this returns nil I think, not sure if we should error or just render an empty data field in the response...